src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala

```diff
@@ -318,27 +318,22 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
 
       if (!settings.docsourceurl.isDefault)
         inSource.map { case (file, line) =>
-          // file path is relative to source root (-sourcepath)
-          val src = Paths.get(settings.sourcepath.value).toUri
-          val path = file.file.toPath.toUri
-          val filePathExt = src.relativize(path)
-          val rawPath: String = filePathExt.getRawPath
+          val filePathExt = {
+            // file path is relative to source root (-sourcepath)
+            val sp = settings.sourcepath.value
+            val fileUri = file.file.toPath.toUri
+            // if the sourcepath is empty use an absolute path
+            if (sp.isEmpty) fileUri.getRawPath
+            else Paths.get(sp).toUri.relativize(fileUri).getRawPath
```

Comment:
In 2.13.11, without `-sourcepath`, the path string is what the compiler got on the command line. So `scaladoc a/B.scala` would insert a relative path.

https://github.com/scala/scala/blob/v2.13.11/src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala#L320

Assuming build tools use absolute paths, I went for an absolute path if `-sourcepath` is empty.

2.13.12 defaults to path relative to the current directory because `Paths.get("").toUri` is the current directory

https://github.com/scala/scala/blob/v2.13.12/src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala#L322

But I think we can use the absolute path in this case.